testEnv_test.go

```diff
@@ -54,21 +59,39 @@ func TestStartTestEnv(t *testing.T) {
 		args    args
 		wantErr bool
 	}{
+		//Only ever have one case here, otherwise test will fail with multiple api servers trying to start
```

Comment:
It does.  The services exit on their own when the program ends (although there is an explicit testEnv.Stop call to terminate them earlier), and I pass back the tempdir so that it can be deleted  (you'll see I defer the removeall after the install).  From my testing it is not possible to have two apiservers running at the same time - they appear to bind to random ports, but the second one never comes up... I think etcd might be colliding on ports in the background.  Having multiple cases could work, but would need serial execution enforcement , which can only be done at cli level.

Reply:
What makes this so? "which can only be done at cli level"

Probably not worth doing now but wonder how you'd parallelise then. But also I'm just curious.

Reply:
the tests will run in parallel by default with whatever level of concurrency go is set to operate at by default.  You can override that behaviour with `run test -p=1` to enforce serial execution, but it's not configurable in the test itself.  I would guess there's something in the etcd config which is not happy with two etcd services running at once, but haven't dug into what that might be.  For most purposes I think just staging one testenv at the start of a testrun will be sufficient and we can adjust which version we're using over time or through ci pipelines if we want parallelism